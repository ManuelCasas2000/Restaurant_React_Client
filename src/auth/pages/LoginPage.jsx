import { Link } from 'react-router-dom'
import { Link as RouterLink } from 'react-router-dom';
import { useForm, useAuthStore } from '../../calendar';
import './LoginPage.css';

const loginFormFields = {
    loginEmail: '',
    loginPassword: ''
}

export const LoginPage = () => {
    const { startLogin, errorMessage } = useAuthStore();
    const {loginEmail, loginPassword, onInputChange: onLoginInputChange} = useForm(loginFormFields);
    
    const loginSubmit = (event) => {
        event.preventDefault();
        console.log({loginEmail, loginPassword});
        startLogin({email:loginEmail, password:loginPassword});
    }

  return (
    <div className="container mt-5 d-flex justify-content-center">
        <div className="col-md-6 login-form-1">
            <div className="row">
                  <h3>Ingreso</h3>
                  <form onSubmit={loginSubmit}>
                      <div className="form-group mb-2">
                          <input 
                              type="text"
                              className="form-control"
                              placeholder="Correo"
                              name="loginEmail"
                              value = {loginEmail}
                              onChange = {onLoginInputChange}
                          />
                      </div>
                      <div className="form-group mb-2">
                          <input
                              type="password"
                              className="form-control"
                              placeholder="ContraseÃ±a"
                              name="loginPassword"
                              value={loginPassword}
                              onChange={onLoginInputChange}
                          />
                      </div>
                      <div className="d-grid gap-2">
                          <input
                              type="submit"
                              className="btnSubmit"
                              value="Login"
                            />
                      </div>
                      <div className="d-flex justify-content-end">
                          <Link component= {RouterLink}  to="/auth/register">
                            Crear una cuenta
                          </Link>
                      </div>
                  </form>
            </div>
        </div>
    </div>
  )
}
