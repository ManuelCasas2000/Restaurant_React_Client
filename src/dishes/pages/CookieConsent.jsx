import React from 'react'
import { useCookies } from 'react-cookie';
import { addDays, addMinutes } from 'date-fns';

export const CookieConsent = () => {
    const [cookies, setCookie] = useCookies(["cookieConsent"]);
    
    const giveCookieConsentAll = () => {
      // Indicando path: "/" esta cookie puede ser accesible desde todas las páginas
      setCookie("cookieConsent", true, {expires: addDays(new Date(),30)}, { path: "/" });
    };

    const giveCookieConsentNecesary = () => {
      // setCookie("cookieConsent", true, {expires: addDays(new Date(),30)}, {path: "http://mirestaurant.com"});
      // Para las pruebas, le pongo una duración de 1 minuto
      setCookie("cookieConsent", true, {expires: addMinutes(new Date(),1)}, {path: "http://mirestaurant.com"});
    }
  
    return (
        <div className="cookies animate__animated animate__backInUp">
    <h4 className="cookies__titulo">¿Aceptas nuestras Cookies?</h4> 
    <p className="cookies__texto">                Utilizamos cookies propias y de terceros con finalidades analíticas y para mostrarte publicidad relacionada con tus 
                preferencias y basada en tus hábitos de navegación y tu perfil. Puedes pulsar "Sólo las necesarias”
                si sólo quieres que usemos las cookies imprescindibles o "Todas" si quieres que todos puedan seguir tus pasos en internet. 
                Para más información, <a href={"https://www.tokioschool.com/politica-de-cookies/"} target="_blank">pulsa en Política de cookies aqui</a> o en el pie de página.
    </p>
    <div className="cookies__botones">
        <button className="btn btn-secondary cookies__boton" onClick={giveCookieConsentNecesary}>Sólo las necesarias</button>
        <button className="btn btn-secondary cookies__boton" onClick={giveCookieConsentAll}>Todas</button>
    </div>
    </div>
    );
}
