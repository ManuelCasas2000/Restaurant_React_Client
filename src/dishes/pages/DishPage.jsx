import { useMemo, useState } from "react";
import { useLocation, useNavigate, useParams } from "react-router-dom"
import Swal from "sweetalert2";
import { useAuthStore } from "../../calendar/hooks/useAuthStore";
import { getDishById } from "../helpers/getDishById";
import { LoadingIcon } from "../components/LoadingIcon";
import {changePage} from "../../store/ui/pageSlice";
import { useDispatch, useSelector } from "react-redux";
import { useEffect } from "react";


export const DishPage = () => {
    const {dishId, currentPage, ...rest} = useParams();

    const {dish}= getDishById(dishId);
    const navigate = useNavigate();
    const {status, checkAuthToken} = useAuthStore();
    const {addDishe, errorMessage} = useAuthStore();
    const isAuthenticated = useMemo(() => status === 'authenticated', [status]);
    const dispatch = useDispatch();
    const {pageGo} = useSelector(state => state.currentPage);

  dispatch(changePage(parseInt(currentPage, 10)));
  
    const onNavigateBack = () => {
      navigate(-1, {
        replace:true
      })
    }

    const onSaveDishe = () => {
      Swal.fire('Añadido correctamente', `El plato '${dish.data.name}' se ha añadido a la reserva`, 'success');
      addDishe({dish:dish.data.name});
    }

    const [dishName, setDishName] = useState({
      idDish: dish.id,
      numberDish: ''
    });

    if(!dish) {
      return <navigate to="/pasta"/>
    }
    return (
      <div>
        { 
          dish.isLoading
          ?(<LoadingIcon/>)
          :(
            <div>
            <div className="row mt-5 animate__animated animate__fadeIn">
              <div className="col-4">
                <div>
                  <img src={dish.data.image} alt={dish.data.name} className="img-thumbnail" />
                </div>
                <br />
              <div className="card mt-5 searchM">
                  <button 
                          disabled = {!isAuthenticated}
                          className="botoncard mt-2 mb-2 ms-5 btn btn-secondary btn-lg"
                          onClick={onSaveDishe}
                          >
                          Select
                        </button>
                  <button
                    className="botoncard mt-2 mb-2 ms-2 btn btn-outline-secondary  btn-lg"
                    onClick={onNavigateBack}
                    >
                    Back
                  </button>
              </div>
              </div>
              <div className="card col-8">
                <br />
                <h3>{dish.data.name}</h3>
                <p className="card-text">{dish.data.instructions}</p>
                <br />
                <h5>Ingredients</h5>
                <p className="card-text">{dish.data.ingredients}</p>
                <h5>It contains gluten </h5>
                <p className="card-text">{dish.data.gluten}</p>
                <h5>Price </h5>
                <p className="card-text">{dish.data.price}€</p>
                <br />
              </div>
            </div>
            <br /><br /><br /><br />
            </div>
          )
        }
      </div>
    )
  }
  