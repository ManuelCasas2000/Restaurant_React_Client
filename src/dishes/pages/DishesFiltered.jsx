import { DishCard } from "../components/DishCard"

export const DishesFiltered = ({results}) => {
  return (
    <div>
        <ul>{results.map((text) => (
                    <DishCard key={text} text={text}></DishCard>
                ))}
            </ul>
    </div>
  )
}
