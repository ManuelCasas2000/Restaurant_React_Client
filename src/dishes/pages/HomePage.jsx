
import '../../styles.css'

export const HomePage = () => {
    
  return (

    <>
    <div className="container mt-6 mb-10 d-flex justify-content-center">
    <div className="col-md-10">
    <div className="alert alert-dark mt-5 animate__animated animate__fadeIn"><h1>Similique optio rem commodi explicabo</h1></div><br />
        <div id="carouselExampleCaptions" className="carousel slide carousel-fade imagenes_carrousel" data-bs-ride="carousel">
            <div className="carousel-indicators">
                <button type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide-to="0" className="active" aria-current="true" aria-label="Slide 1"></button>
                <button type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide-to="1" aria-label="Slide 2"></button>
                <button type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide-to="2" aria-label="Slide 3"></button>
                <button type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide-to="3" aria-label="Slide 4"></button>
            </div>
            <div className="carousel-inner">
                <div className="carousel-item active" data-bs-interval="2000">
                    <img src="http://res.cloudinary.com/dla3ofowl/image/upload/v1679923849/restaurant/q0nmdiu5iqrrgmvcmv4a.jpg" className="d-block w-100" style={{ height: 'calc(90vh - 90px)'  }}/>
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Lorem, ipsum dolor sit amet consectetur adipisicing elit. Similique optio rem commodi explicabo, iste recusandae facere ab quis neque ipsum laudantium iusto, enim ex, alias dolore nostrum nemo cupiditate accusantium?</h5>
                    </div>
                </div>
                <div className="carousel-item"  data-bs-interval="2000">
                    <img src="http://res.cloudinary.com/dla3ofowl/image/upload/v1679923997/restaurant/dd2bwtczkl7fsmdjn2rt.jpg" className="d-block w-100" style={{ height: 'calc(90vh - 90px)' }}/>
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Nulla minus quaerat perferendis veniam est voluptates, accusamus sed. Dolor ratione totam itaque autem laudantium. Officia dolore praesentium cupiditate reprehenderit tempora delectus?.</h5>
                    </div>
                </div>
                <div className="carousel-item"  data-bs-interval="2000"> 
                    <img src="http://res.cloudinary.com/dla3ofowl/image/upload/v1679924082/restaurant/pysdpwg4pfojxykrxbwu.jpg" className="d-block w-100" style={{ height: 'calc(90vh - 90px)' }}/>
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Lorem ipsum dolor sit amet consectetur adipisicing elit. Rerum, ratione repellat modi hic minus atque nostrum rem est libero debitis reiciendis nam perferendis aliquam nemo necessitatibus porro facere architecto? Dolorem?.</h5>
                    </div>
                </div>
                <div className="carousel-item">
                    <img src="http://res.cloudinary.com/dla3ofowl/image/upload/v1679924154/restaurant/vrjswkvznxab26se4tw5.jpg" className="d-block w-100" style={{ height: 'calc(90vh - 90px)' }}/>
                    <div className="carousel-caption d-none d-md-block">
                        <h5>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quis perspiciatis sequi odio magni doloremque eius! Nihil inventore totam cum eius non. Dolores, minima? Qui cupiditate aut veritatis corrupti minima veniam!.</h5>
                    </div>
                </div>
            </div>
            <button className="carousel-control-prev" type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide="prev">
                <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                <span className="visually-hidden">Previous</span>
            </button>
            <button className="carousel-control-next" type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide="next">
                <span className="carousel-control-next-icon" aria-hidden="true"></span>
                <span className="visually-hidden">Next</span>
            </button>
        </div>
        <div className='mt-5'>
            <h5>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Vel dolor nam provident soluta pariatur itaque saepe rem odit quasi ea quis tenetur, quidem, quae modi odio tempore repellat eos quaerat. Lorem, ipsum dolor sit amet consectetur adipisicing elit. Tempore, labore error magnam.</h5>
        </div>
        <div className='mt-3'>
            <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Deserunt dolor voluptates consectetur magni dolores, laborum ea amet minus totam nihil tempora! Consectetur incidunt non sed, inventore magni assumenda? Illo, veritatis? Lorem ipsum dolor sit, amet consectetur adipisicing elit. Vel dolor nam provident soluta pariatur itaque saepe rem odit quasi ea quis tenetur, quidem, quae modi odio tempore repellat eos quaerat. Lorem, ipsum dolor sit amet consectetur adipisicing elit. Tempore, labore error magnam, delectus in animi vero voluptas iste, fugit libero deleniti laudantium ea ab. Facere maiores quas voluptates veniam ducimus.</p>
        </div>
        <div className='mt-2'>
            <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Provident voluptatibus deserunt amet, illo molestiae atque non nesciunt dolores numquam consectetur odio aut perferendis pariatur odit doloribus a natus repellendus nulla? Lorem ipsum dolor sit amet consectetur adipisicing elit. Deserunt dolor voluptates consectetur magni dolores, laborum ea amet minus totam nihil tempora! Consectetur incidunt non sed, inventore magni assumenda? Illo, veritatis? Lorem ipsum dolor sit, amet consectetur adipisicing elit. Vel dolor nam provident soluta pariatur itaque saepe rem odit quasi ea quis tenetur, quidem, quae modi odio tempore repellat eos quaerat. Lorem, ipsum dolor sit amet consectetur adipisicing elit. Tempore, labore error magnam, delectus in animi vero voluptas iste, fugit libero deleniti laudantium ea ab. Facere maiores quas voluptates veniam ducimus.</p>
        </div>
    </div>
    </div>
    <br />
    <br />
    </>
  )
}
