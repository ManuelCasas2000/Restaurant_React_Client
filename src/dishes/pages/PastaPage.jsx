import { DishList } from "../components/DishList"

export const PastaPage = () => {
    return (
        <>
            <div className="alert alert-dark mt-5 animate__animated animate__fadeIn"><h1>These are our pasta dishes</h1></div><br />
            <hr />
            <DishList cathegory='Pasta'/>
            <br /><br />
        </>
    )
}
