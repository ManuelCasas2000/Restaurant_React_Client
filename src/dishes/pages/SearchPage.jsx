import { useLocation, useNavigate } from 'react-router-dom';
import {useForm} from '../hooks/useForm';
import queryString from 'query-string';
import { getDishByName } from '../helpers/getDishByName';
import { DishCardFile } from '../components/DishCardFile';

export const SearchPage = () => {

  const navigate = useNavigate();
  const location = useLocation();
  const {q=''} = queryString.parse(location.search);

  const dishes = getDishByName(q);

  const showSearch = (q.length === 0);
  const showError = (q.length > 0) && (dishes.length === 0);

  
  const {searchText, onInputChange} = useForm({
    searchText: q
  })

  const onSearchSubmit = (event) => {
    event.preventDefault();
    if(searchText.trim().length <= 2) return;
    navigate(`?q=${searchText.toLowerCase().trim()}`);
  }


  return (
    <>
      <div className="alert alert-dark mt-5 animate__animated animate__bounceInDown"><h1>Search your dishes by name</h1></div><br />
      <hr />
      <div className="row  animate__animated animate__fadeIn">
        <div className="col-4">
          <form onClick={onSearchSubmit} aria-label="form">
            <input 
              type="text" 
              placeholder="Search a dish"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={onInputChange} 
            />
            <button className="btn btn-outline-secondary mt-3">Search</button>
          </form>
        </div>
        <div className="col-8">
          <div 
            className="alert alert-secondary animate__animated animate_fadeIn"
            style={{display: showSearch ? '': 'none'}}  
          >
          </div>
          <div
            aria-label="no-result" 
            className="alert alert-danger animate__animated animate_fadeIn"
            style={{display: showError ? '': 'none'}}
            >
            There is not results with <b>{q}</b>
          </div>
          {
            dishes.map(dish => (
              // <DishCard key={dish.id} {...dish}/>
              <DishCardFile key={dish.id} {...dish}/>
            ))
          }
        </div>
          <div id="altura-search"></div>
      </div>
    </>
  )
}
