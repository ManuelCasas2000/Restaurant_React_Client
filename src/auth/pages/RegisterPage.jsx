
import { Link } from 'react-router-dom';
import { Link as RouterLink } from 'react-router-dom';
import Swal from 'sweetalert2';
import { useForm, useAuthStore } from '../../calendar';
import './LoginPage.css';


const formData = {
  email:'manuel@correo.es',
  password:'1234',
  displayName:'Manuel Casas',
}

const registerFormFields = {
    registerName: '',
    registerEmail: '',
    registerPassword: '',
    registerPassword2: ''
}

export const RegisterPage = () => {

    const {startRegister,errorMessage} = useAuthStore();
    const {registerEmail, registerName, registerPassword, registerPassword2, onInputChange: onRegisterInputChange} = useForm(registerFormFields);

  const registerSubmit = (event) => {
    event.preventDefault();
    if(registerPassword !== registerPassword2) {
        Swal.fire('Error en el registro','Las contraseñas son diferentes.','error');
        return;
    }
    console.log({registerEmail, registerName, registerPassword, registerPassword2});
    startRegister({name:registerName, email:registerEmail, password:registerPassword});
  }

    return (
      <div className="container mt-5 d-flex justify-content-center">
        <div className="col-md-6 login-form-1">
                  <h3>Registro</h3>
                  <form onSubmit={registerSubmit}>
                      <div className="form-group mb-2">
                          <input
                              type="text"
                              className="form-control"
                              placeholder="Nombre"
                              name="registerName"
                              value={registerName}
                              onChange={onRegisterInputChange}
                          />
                      </div>
                      <div className="form-group mb-2">
                          <input
                              type="email"
                              className="form-control"
                              placeholder="Correo"
                              name="registerEmail"
                              value={registerEmail}
                              onChange={onRegisterInputChange}
                          />
                      </div>
                      <div className="form-group mb-2">
                          <input
                              type="password"
                              className="form-control"
                              placeholder="Contraseña" 
                              name="registerPassword"
                              value={registerPassword}
                              onChange={onRegisterInputChange}
                          />
                      </div>

                      <div className="form-group mb-2">
                          <input
                              type="password"
                              className="form-control"
                              placeholder="Repita la contraseña" 
                              name="registerPassword2"
                              value={registerPassword2}
                              onChange={onRegisterInputChange}
                          />
                      </div>

                      <div className="d-grid gap-2">
                          <input 
                              type="submit" 
                              className="btnSubmit" 
                              value="Crear cuenta"
                            />
                      </div>
                      <div className="d-flex justify-content-end">
                          <Link component= {RouterLink} to="/auth/login">
                            Ingreso
                          </Link>
                      </div>
                  </form>
              </div>
        </div>

    )
  }