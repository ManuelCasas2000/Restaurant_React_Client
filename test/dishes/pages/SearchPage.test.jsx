import {render, screen} from '@testing-library/react';
import { MemoryRouter} from 'react-router-dom';
import { SearchPage } from '../../../src/dishes/pages/SearchPage';


describe('Pruebas en <SearchPage/>', () => {

    test('Debe de mostrarse correctamente con valores por defecto', () => {
        //Con MemoryRouter indentificamos en qué url estoy
        const {container} = render(
            <MemoryRouter> 
                <SearchPage/>
            </MemoryRouter>
        );
        //screen.debug();
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar -Chilli- y el input con el valor del queryString', () => {
        //Con MemoryRouter indentificamos en qué url estoy
        render(
            <MemoryRouter initialEntries={['/search?q=chilli']}> 
                <SearchPage/>
            </MemoryRouter>
        );

        // <input
        // ...
        //     value="chilli"

        const input = screen.getByRole('textbox');
        expect(input.value).toBe('chilli');

        // <img
        // ...
        // src="http://res.cloudinary.com/dla3ofowl/image/upload/v1677168482/restaurant/qitsotv32rv7wqc2xksq.jpg"

        const img = screen.getByRole('img');
        expect(img.src).toContain('qitsotv32rv7wqc2xksq');

        //     <div
        //     aria-label="no-result" 
        // ...
        //     There is not results with <b>{q}</b>

        const noResult = screen.getByLabelText('no-result');

        // console.log(noResult.style);
        // CSSStyleDeclaration {
        // ...
        //     _values: { display: 'none' },

        expect(noResult.style.display).toBe('none');
    });

    test('Debe mostrar un error si no se encuentra el plato (chilliNoExite)', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=chilliNoExite']}> 
                <SearchPage/>
            </MemoryRouter>
        );
        const noResult = screen.getByLabelText('no-result');
        expect(noResult.style.display).toBe('');
    });

});

