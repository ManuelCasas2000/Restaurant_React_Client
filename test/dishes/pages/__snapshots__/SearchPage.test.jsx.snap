// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="alert alert-dark mt-5 animate__animated animate__bounceInDown"
  >
    <h1>
      Search your dishes by name
    </h1>
  </div>
  <br />
  <hr />
  <div
    class="row  animate__animated animate__fadeIn"
  >
    <div
      class="col-4"
    >
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a dish"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-secondary mt-3"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-8"
    >
      <div
        class="alert alert-secondary animate__animated animate_fadeIn"
      />
      <div
        aria-label="no-result"
        class="alert alert-danger animate__animated animate_fadeIn"
        style="display: none;"
      >
        There is not results with 
        <b />
      </div>
    </div>
    <div
      id="altura-search"
    />
  </div>
</div>
`;
